kstore-file/src/commonMain/kotlin/io/github/xxfast/kstore/file/format/KStoreFormat.kt

```diff
@@ -0,0 +1,23 @@
+package io.github.xxfast.kstore.file.format
+
+import kotlinx.io.Sink
+import kotlinx.io.Source
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationStrategy
+
+public abstract class KStoreFormat<T : @Serializable Any>(public val serializer: KSerializer<T>) {
```

Comment:
Is there a reason why we need this intermediary class instead of passing in the serialiser directly?

Reply:
Not all formats support KotlinX.IO

Reply:
as before you do not need to change this, since there is a JSON serializer, this is simply done if you really need to change this, kotlinx.io allows you to convert this to java bytearray and/or string

Reply:
Yeah prolly best if we do this separately - and I agree we would need some abstraction to handle multiple formats, but we will cross that bridge when we get there